### PR TITLE
feat: add support for template-generator versioning requirements

### PR DESCRIPTION
**Pull request by fmvilas**
**Description**
It allows template owners to specify which version range of the generator they support. For instance, if we find a `.tp-config.json` file with the attribute `"generator": ">=0.36.0 <=1.0.0"` and the current version of the generator is `1.5.0`, it should trigger an error.

The format of the `generator` config property is a string representing semantic versioning ranges. See [semver](https://www.npmjs.com/package/semver) for more details.

**Related issue(s)**
Partially solves #280

### CODE REVIEW

File: lib/generator.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const Ajv = require('ajv');
 const filenamify = require('filenamify');
 const git = require('simple-git/promise');
 const npmi = require('npmi');
+const semver = require('semver');
+const packageJson = require('../package.json');
```
Comment (derberg) [inline lib/generator.js:17]: I think better would be to read the version from environment variable `process.env.npm_package_version` as we do it here already https://github.com/asyncapi/generator/blob/master/package.json#L18

Comment (fmvilas) [reply lib/generator.js:17]: Why is this approach better than reading from the package.json file? It's not obvious that this environment variable exists. Also, if someone uses Yarn (or another package manager), does this environment variable exist too?

**Comment by fmvilas** [comment]
> It would be a nice DX if we could enrich the info displayed in the console with information what is the latest version of the template compatible with a given version of the generator

This is hard to implement. I can offer the opposite: the latest generator version that works with the template. How do we find the latest version of the template compatible with the current version of the generator? Any ideas?

File: lib/generator.js (hunk 1 above)
Comment (derberg) [reply lib/generator.js:17]: This is officially supported variable https://docs.npmjs.com/misc/scripts#packagejson-vars

At the end is it alwasy better to not read file if not needed.

Yarn also supports it https://classic.yarnpkg.com/en/docs/cli/version/#toc-version-lifecycle

Comment (fmvilas) [reply lib/generator.js:17]: I don't think this environment variable is available outside `preversion`, `version`, and `postversion`.

Screen Shot 2020-04-28 at 11 15 02

**Comment by derberg** [comment]
@fmvilas I'm not saying it is easy, I'm just saying it is better :) we can have a separate issue for it for further discussion and research. At least official templates should have such functionality

File: lib/generator.js (hunk 1 above)
Comment (fmvilas) [reply lib/generator.js:17]: Definitely they're not available. Check docs here: https://docs.npmjs.com/misc/scripts#environment. It's only when running a package script.

Comment (derberg) [reply lib/generator.js:17]: ok, forget it, it is too risky, works only if you run code with `npm` script, it will only complicate the development flow

**Comment by derberg** [comment]
@fmvilas just update docs and I approve

**Comment by fmvilas** [comment]
Docs updated 😊

**Pull request merged**